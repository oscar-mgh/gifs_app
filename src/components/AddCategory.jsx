import PropTypes from 'prop-types';
import {useState} from 'react';

export const AddCategory = ({setCategories, onNewCategory}) => {
	const [inputValue, setInputValue] = useState('');

	const onInputChange = ({target}) => {
		setInputValue(target.value);
	};

	const onSubmit = (e) => {
		e.preventDefault();
		if (inputValue.trim().length > 1) {
			onNewCategory(inputValue.trim());
			setInputValue('');
		}
	};

	return (
		<form onSubmit={onSubmit} aria-label='form'>
			<input
				type='text'
				placeholder='Search gifs'
				value={inputValue}
				onChange={onInputChange}
			/>
		</form>
	);
};

AddCategory.propTyes = {
	onNewCategory: PropTypes.func.isRequired,
};
