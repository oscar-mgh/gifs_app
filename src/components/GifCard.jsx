import PropTypes, {string} from 'prop-types';

export const GifCard = ({url, title}) => {
	return (
		<div className='card'>
			<h2>{title}</h2>
			<img src={url} alt={title} />
		</div>
	);
};

GifCard.propTypes = {
	title: PropTypes.string.isRequired,
	url: PropTypes.string.isRequired,
};
