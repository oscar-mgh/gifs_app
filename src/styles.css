* {
	font-family: Helvetica, Arial, sans-serif;
	background-color: rgb(230, 230, 241);
}

body {
	padding: 60px;
}

input {
	background-color: white;
	border-radius: 5px;
	border: 1px solid rgb(0, 110, 255);
	color: black;
	font-size: 1.4rem;
	outline: none;
	padding: 10px 15px;
	width: 100%;
}

.App__title {
	margin: 0 0 4rem 0;
	text-align: center;
	font-size: 3.2rem;
	color: rgb(0, 25, 59);
	font-weight: 700;
}

h2 {
	text-align: center;
	font-size: 1.4rem;
	background-color: transparent;
}

h3 {
	font-size: 2.5rem;
	margin-left: 30px;
	margin-bottom: 30px;
}

.card-grid {
	display: flex;
	flex-direction: row;
	flex-wrap: wrap;
	justify-content: center;
}

.card {
	min-height: 400px;
	max-height: 500px;
	min-width: 400px;
	max-width: 400px;
	align-content: center;
	align-items: center;
	background-color: rgb(247, 247, 247);
	border-radius: 10px;
	box-shadow: 0px 3px 5px rgba(0, 0, 0, 0.03);
	display: flex;
	flex-direction: column;
	height: 0%;
	justify-content: center;
	margin-bottom: 25px;
	margin-right: 25px;
	overflow: visible;
}

.card p {
	background-color: white;
	flex: 1;
	font-size: 1.5rem;
	margin-top: 5px;
	padding: 5px 20px 0px 20px;
	text-align: center;
}

.card img {
	width: 100%;
}
