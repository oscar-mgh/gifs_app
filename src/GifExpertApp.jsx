import {useState} from 'react';
import {AddCategory, GifGrid} from './components';

export const GifExpertApp = () => {
	const [categories, setCategories] = useState(['SUPER SAIYAN BLUE']);

	const onAddCategory = (newCategory) => {
		if (categories.includes(newCategory)) return;
		setCategories([newCategory ,...categories]);
	};

	return (
		<div>
			<h1 className='app__title'> Gif App </h1>

			<AddCategory onNewCategory={onAddCategory} />

			{categories.map((category) => (
				<GifGrid key={category} category={category} />
			))}
		</div>
	);
};
