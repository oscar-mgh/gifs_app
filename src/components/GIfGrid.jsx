import {GifCard} from './GifCard';
import {useFetchGifs} from '../hooks/useFetchGifs';
import PropTypes from 'prop-types';

export const GifGrid = ({category}) => {
	const {images, isLoading} = useFetchGifs(category);

	return (
		<div>
			<h3>{category}</h3>

			{isLoading && <h2> Loading ...</h2>}

			<div className='card-grid'>
				{images.map((image) => (
					<GifCard key={image.id} {...image} />
				))}
			</div>
		</div>
	);
};

GifGrid.propTypes = {
	category: PropTypes.string.isRequired,
};
