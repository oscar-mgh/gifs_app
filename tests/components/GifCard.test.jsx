import { render, screen } from "@testing-library/react"
import { GifCard } from '../../src/components';

describe('Pruebas en <GifCard />', () => {

  const title = "Roku"
  const url = "https://roku.com/avatar"

  test('Debe hacer match con el snapshot ', () => {
    
    const { container } = render( <GifCard title={title} url={url} /> )
    expect( container ).toMatchSnapshot()
  
  });
  
  test('Debe de mostrar la imagen con el URL y el ALT indicado ', () => {
    
    render( <GifCard title={title} url={url} /> )
    const { src, alt } = screen.getByRole('img')
    expect(src).toBe(url)
    expect(alt).toBe(title)
  
  });

  test('Debe de mostrar el titulo en el componente', () => {
    
    render( <GifCard title={title} url={url} /> )
    expect(screen.getByText(title)).toBeTruthy()

  });

});