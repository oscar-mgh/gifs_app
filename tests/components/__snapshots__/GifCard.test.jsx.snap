// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifCard /> Debe hacer match con el snapshot  1`] = `
<div>
  <div
    class="card"
  >
    <h2>
      Roku
    </h2>
    <img
      alt="Roku"
      src="https://roku.com/avatar"
    />
  </div>
</div>
`;
